src/core.jl

```diff
@@ -141,8 +213,14 @@ function _test_reference(equiv, rendermode, file::File, actual::T) where T
     end
 
     # file exists
+    actual = preprocess(file, actual; kw...)
     reference = loadfile(T, file)
 
+    if isnothing(equiv)
+        # generally, `reference` and `actual` are of the same type after preprocessing
+        equiv = default_equality(typeof(reference), typeof(actual))
+    end
+
```

Comment:
infer the default equality method after preprocessing because now `reference` and `actual` have the same type.